tests/functional/object/mcg/test_internal_route_auth.py

```diff
@@ -62,79 +140,137 @@ def test_mgmt_and_endpoint_route_authentication(self):
 
         # Validate noobaa endpoint host address + port without using auth token
-        response = requests.get(f"http://{endpoint_host_ip}:{port}", verify=False)
-        logger.info(response.status_code)
+        response_code = self.acccess_url(ep_url, verify=False)
         assert (
-            response.status_code == 401
-        ), f"Expected response code 401. Received {response.status_code}"
+            response_code == 401
+        ), f"Expected response code 401. Received {response_code}"
 
         # Validate noobaa endpoint host address + port with incorrect auth token
-        response = requests.get(
-            f"http://{endpoint_host_ip}:{port}",
-            headers=invalid_headers,
+        response_code = self.acccess_url(ep_url, header=invalid_headers, verify=False)
+        assert (
+            response_code == 403
+        ), f"Expected response code 403. Received {response_code}"
+
+    @pytest.mark.parametrize(
+        argnames="flag",
+        argvalues=[
+            pytest.param("NOOBAA_VERSION_AUTH_ENABLED"),
+            pytest.param("NOOBAA_METRICS_AUTH_ENABLED"),
+        ],
+        ids=[
+            "NOOBAA_VERSION_AUTH_ENABLED",
+            "NOOBAA_METRICS_AUTH_ENABLED",
+        ],
+    )
+    @tier2
+    def test_noobaa_configmap_flag(self, reset_configmap, flag):
+        """
+        This test covers validation of the below steps
+            1. Validate below values are set to true in noobaa-config configmap
+                 i. NOOBAA_VERSION_AUTH_ENABLED: True
+                ii. NOOBAA_METRICS_AUTH_ENABLED: True
+            2. Validate respective routes using auth token
+            3. Change flag value to "False"
+            4. Validate respective routes without using auth token
+            5. revert value to original
+        """
+
+        json_web_token = get_noobaa_metrics_token_from_secret()
+        auth_headers = {"Authorization": f"Bearer {json_web_token}"}
+        if flag == "NOOBAA_VERSION_AUTH_ENABLED":
+            test_url = self.get_mgmt_route_address()
+        else:
+            test_url = self.get_endpoint_host_ip_port()
+
+        # Validate given flag value is set to true in noobaa-config configmap
+        noobaa_configmap = OCP(
+            kind="configmap",
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+            resource_name=constants.NOOBAA_CONFIGMAP,
+        )
+        cm_data = noobaa_configmap.get()["data"]
+        assert (
+            cm_data.get(flag) == "true"
+        ), f"Incorrect default value of {flag}. Received value {cm_data.get(flag)}"
+
+        # Validate respective routes using auth token
+        response_code = self.acccess_url(test_url, header=auth_headers, verify=False)
+        assert (
+            response_code == 200
+        ), f"Expected response code 200. Received {response_code}"
+
+        # Change flag value to "False"
+        self.change_config_flag(flag)
+        import time
```

Comment:
Please move this import to the top of the file if possible

Reply:
Fixed